collections/collections.go

```diff
@@ -0,0 +1,98 @@
+package collections
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// StorageProvider represents sdk.Context
+// it is used to avoid to reduce dependencies.
+type StorageProvider interface {
+	// KVStore returns a KVStore given its StoreKey.
+	KVStore(key storetypes.StoreKey) storetypes.KVStore
+}
+
+// ErrNotFound defines an error returned when
+// a key was not found.
+type ErrNotFound struct {
```

Comment:
Typically you want `ErrFoo` to identify error _values_ e.g. `var ErrFoo = errors.New("foo")`, and for error types like these to be named as e.g. `NotFoundError`. Also, you want to define methods on error types on pointer receivers, for complicated and annoying reasons.